File: app/src/main/java/com/br/example/fakebank/presentations/viewModels/viewModelFactories/LoginViewModelFactory.java
package com.br.example.fakebank.presentations.viewModels.viewModelFactories;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.br.example.fakebank.domains.services.LoginService;
import com.br.example.fakebank.presentations.handles.LoginHandle;
import com.br.example.fakebank.presentations.viewModels.LoginViewModel;

public class LoginViewModelFactory implements ViewModelProvider.Factory {
    private LoginService mainService;
    private LoginHandle loginHandle;

    public LoginViewModelFactory(LoginService mainService, LoginHandle loginHandle) {
        this.mainService = mainService;
        this.loginHandle = loginHandle;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(LoginViewModel.class)){
            return (T) new LoginViewModel(mainService, loginHandle);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: BankApp/app/src/main/java/com/paulokeller/bankapp/ui/login/LoginViewModelFactory.kt
package com.paulokeller.bankapp.ui.login

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.paulokeller.bankapp.data.repositories.Repository
import com.paulokeller.bankapp.data.services.Client

class LoginViewModelFactory(private val repository: Repository, private val client: Client) : ViewModelProvider.NewInstanceFactory()  {
    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return LoginViewModel(repository, client) as T
    }
}